Sources/MongoSwift/Operations/MongocCommandHelpers.swift

```diff
@@ -0,0 +1,58 @@
+import CLibMongoC
+
+/// Executes the provided closure using a stack-allocated, mutable bson_t. The bson_t is only valid for the body of the
+/// closure and must be copied if you wish to use it later on.
+internal func withStackAllocatedMutableBSONPointer<T>(body: (MutableBSONPointer) throws -> T) rethrows -> T {
+    var bson = bson_t()
+    defer {
+        withUnsafeMutablePointer(to: &bson) { ptr in
```

Comment:
as far as I can tell, there is no functional difference between using this `withUnsafeMutablePointer` helper and just using `&` directly when interoperating with C methods. however, I went with this approach as it seems more explicit.